.github/workflows/test.yml

```diff
@@ -0,0 +1,48 @@
+name: Test
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  push:
+    branches:
+      - '**'
+env:
+  TERM: xterm-256color
+jobs:
+  php8-compatibility:
+    name: PHP 8.x Compatibility
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: PHP Compatibility
+        uses: pantheon-systems/phpcompatibility-action@dev
+        with:
+          test-versions: 8.0-
+          paths: ${{ github.workspace }}/*.php ${{ github.workspace }}/inc/*.php ${{ github.workspace }}/inc/network/*.php
+  test:
+    needs: php8-compatibility
+    name: Test
+    runs-on: ubuntu-latest
+    services:
+      mariadb:
+        image: mariadb:10.6
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup PHP 8.3
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.3
+          extensions: mysqli, zip, imagick
+      - name: Start MySQL
+        run: sudo systemctl start mysql
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/vendor
+          key: test-phpunit-dependencies-${{ hashFiles('composer.json') }}
+          restore-keys: test-phpunit-dependencies-${{ hashFiles('composer.json') }}
```

Comment:
Is there a restore cache step?

Reply:
possibly not although I'm not positive. This was also largely copy-pasta from HUD.